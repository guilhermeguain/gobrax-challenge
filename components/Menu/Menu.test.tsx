import { render, screen } from "@testing-library/react-native";

import { Menu } from "./";

describe("Component: Menu", () => {
  const links = [
    { href: "/", text: "Motoristas" },
    { href: "/vehicles", text: "VeÃ­culos" },
  ];

  it("renders correctly", () => {
    const { toJSON } = render(<Menu />);

    expect(toJSON()).toMatchSnapshot();
  });

  it("contains all links with correct text", () => {
    render(<Menu />);

    links.forEach(({ text }) => {
      expect(screen.getByText(text)).toBeTruthy();
    });
  });
});
