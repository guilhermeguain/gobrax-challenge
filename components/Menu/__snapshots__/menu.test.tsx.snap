// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Menu renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 32,
    }
  }
>
  <Text
    href="/"
    onPress={[Function]}
    role="link"
    style={
      {
        "color": "#000",
        "fontWeight": "bold",
      }
    }
  >
    <Text>
      Motoristas
    </Text>
  </Text>
  <Text
    href="/vehicles"
    onPress={[Function]}
    role="link"
    style={
      {
        "color": "#000",
        "fontWeight": "bold",
      }
    }
  >
    <Text>
      Ve√≠culos
    </Text>
  </Text>
</View>
`;
